### PR TITLE
Move E2E tests to main test.yml workflow

### PR DESCRIPTION
**Pull request by mehulkar**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -213,6 +213,34 @@ jobs:
 
       - name: Integration Tests
         run: pnpm -- turbo run integration-tests --filter=cli
+
+  go_e2e:
+    name: Go E2E Tests
+    needs: determine_jobs
+    if: needs.determine_jobs.outputs.go == 'true'
+    timeout-minutes: 30
```
Comment (chris-olszewski) [inline .github/workflows/test.yml:221]: It looks like this might not be enough time to build and run the tests on Windows 😢

**Review by chris-olszewski — DISMISSED** [review]

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -213,6 +213,34 @@ jobs:
 
       - name: Integration Tests
         run: pnpm -- turbo run integration-tests --filter=cli
+
+  go_e2e:
+    name: Go E2E Tests
+    needs: determine_jobs
+    if: needs.determine_jobs.outputs.go == 'true'
```
Comment (chris-olszewski) [inline .github/workflows/test.yml:220]: We should also run if anything in `crates/turborepo*` changes along with global changes like `Cargo.lock`

File: .github/workflows/test.yml (hunk 1 above)
Comment (mehulkar) [reply .github/workflows/test.yml:221]: it's the same as it is now? https://github.com/vercel/turbo/blob/01e143d80720b532369383e12c60f87368811185/.github/workflows/pr-go-e2e-filtered.yml#L20

File: .github/workflows/test.yml (hunk 2 above)
Comment (mehulkar) [reply .github/workflows/test.yml]: updated and also included prysk tests for this condition if you want to take a look

**Comment by tknickman** [comment]
You'll need to mark these as required as well (like here): https://github.com/vercel/turbo/pull/3519

**Comment by mehulkar** [comment]
@tknickman done

**Review by chris-olszewski — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -82,13 +82,19 @@ jobs:
...
         uses: technote-space/get-diff-action@v6
         with:
           PATTERNS: |
             cli/**
+      - name: Go E2E related changes
+        id: go_e2e
+        uses: technote-space/get-diff-action@v6
+        with:
+          PATTERNS: |
+            shim/**
```
Comment (NicholasLYang) [inline .github/workflows/test.yml:95]: shim is no longer needed. We should probably add `turbo-updater` too

Comment (mehulkar) [reply .github/workflows/test.yml:95]: removing shim, but can we rename turbo-updater to turborepo-updater instead? Why the naming inconsistency?

**Comment by mehulkar** [comment]
Merging this. Rust tests failed, but @jridgewell confirmed that they're failing at the moment. All other checks are green

**Pull request merged**